src/main/resources/com/sonyericsson/jenkins/plugins/bfa/model/FailureCauseBuildAction/summary.groovy

```diff
@@ -31,20 +31,27 @@ def l = namespace(lib.LayoutTagLib)
 
 index = 1
 
-tr {
-    td {
-        img(width: "48", height: "48", src: my.getImageUrl(), style: "margin-right:1em;")
-    }
-    td(style: "vertical-align: middle;") {
-        h2(_("Identified problems"))
-    }
-}
+def rootFailureDisplayData = my.getFailureCauseDisplayData()
+if (! (rootFailureDisplayData.getFoundFailureCauses().empty
+	   && rootFailureDisplayData.getDownstreamFailureCauses().empty
+	   && !PluginImpl.getInstance().isNoCausesEnabled())) {
```

Comment:
puh, I had to write down a truth table to make sure this was correct :) But it seems to be.
Have you done some test scenarios for this so someone don't mess it up in the future?

Could you simplify the logic somehow so it's easier to read?